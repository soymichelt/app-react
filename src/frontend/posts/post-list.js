import React from 'react';

export const PostList = (props) => {
    const {
        data,
        onNext,
        onPrev,
        next,
        prev,
        isOnline
    } = props;

    return (
        <>
            <h1 className='header-title'>Listado de Datos</h1>
            <ul id='list' >
                {data && data.map(item => {
                    return (
                        <li key={item.id}>{item.name}</li>
                    );
                })}
            </ul>
            <button disabled={prev === null ? true : false} onClick={onPrev}>Anterior</button>
            <button disabled={next === null ? true : false} onClick={onNext}>Siguiente</button>
            {isOnline !== null && (
                <p>{isOnline === true ? 'Online' : 'Offline'}</p>
            )}
        </>
    );
};