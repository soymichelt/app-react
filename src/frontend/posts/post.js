import React from 'react';
import './post.scss';

const Post = (props) => {

    const { name, date, avatarUrl, imageUrl, } = props;

    return (
        <div className='post'>
            <div className='post-header'>
                <img className='avatar' src={avatarUrl} />
                <h1 className='header'>
                    <span className='name'>{name}</span>
                    <span className='date'>{date}</span>
                </h1>
            </div>
            <img className='image' src={imageUrl} />
            <footer className='post-footer'>
                <button className='post-action'>Like</button>
                <button className='post-action'>Compartir</button>
                <button className='post-action'>Denunciar</button>
            </footer>
        </div>
    );

};

export default Post;