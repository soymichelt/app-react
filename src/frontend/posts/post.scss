body {
    background-color: #8f57fd;
    font-family: Arial, Helvetica, sans-serif;
}

.post {
    border-radius: 4px;
    background-color: white;
    overflow: hidden;
    position: relative;
    display: inline-block;
    width: 300px;
    height: 280px;
    padding: 8px;
    box-sizing: border-box;
    margin-right: 10px;
    font-size: 20px;
    cursor: pointer;
    transition: 450ms all;
    -webkit-transform-origin: center left;
            transform-origin: center left;
}

.post .post-header {
    margin-bottom: 8px;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.post .header {
    height: 32px;
    line-height: 16px;
    margin: 0;
    padding: 0;
    margin-left: 10px;
}

.post .name {
    font-size: 14px;
}

.post .date::before {
    content: ' â€¢ ';
}

.post .date {
    font-size: 11px;
    color: #999;
}

.post .avatar {
    width: 32px;
    height: 32px;
    object-fit: cover;
}

.post .image {
    width: 300px;
    height: 180px;
    object-fit: cover;
    margin-left: -8px;
    margin-right: -8px;
}

.post .post-footer {
    display: flex;
}

.post .post-action {
    flex: 1;
    font-weight: bold;
    height: 42px;
    border: none;
    background-color: transparent;
    cursor: pointer;
}

.post .post-action:hover {
    color: #505050;
}