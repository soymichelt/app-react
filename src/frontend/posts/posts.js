import React from 'react';
import Post from './post';

const Posts = () => {

    return (
        <Post
            name={'Michael TraÃ±a'}
            date={'25 de diciembre'}
            avatarUrl={'https://soymichel.com/res/circleProfile.png'}
            imageUrl={'https://parzibyte.me/blog/wp-content/uploads/2019/09/forEach-en-JavaScript-Ejemplos-y-sintaxis.png'}
        />
    );

};

export default Posts;