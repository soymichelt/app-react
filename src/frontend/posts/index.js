import React, {useState, useEffect} from 'react';
import {PostList} from './post-list';

export const Posts = () => {
    const [on, setOn] = useState(!navigator ? null : navigator.onLine);
    const [posts, setPosts] = useState([]);
    const [next, setNext] = useState(null);
    const [prev, setPrev] = useState(null);
    const [url, setUrl] = useState('https://rickandmortyapi.com/api/character/');
    const handleNextClick = () => setUrl(next);
    const handlePrevClick = () => setUrl(prev);

    const handleConnectedChanged = () => {
        if(!navigator) setOn(null);
        setOn(navigator.onLine);
    };

    useEffect(() => {
        if(window) {
            window.addEventListener('online', handleConnectedChanged);
            window.addEventListener('offline', handleConnectedChanged);
        }

        const getPosts = async () => {
            try {
                const res = await fetch(url);
                const data = await res.json();
                setPosts(data.results);
                setNext(data.info.next);
                setPrev(data.info.prev);
            }
            catch(e) {
                console.log(e);
            }
        };
        getPosts();

        return () => {
            if(window) {
                window.removeEventListener('online', handleConnectedChanged);
                window.removeEventListener('offline', handleConnectedChanged);
            }
        };
    }, [url]);

    return (
        <PostList
            data={posts}
            onNext={handleNextClick}
            onPrev={handlePrevClick}
            next={next}
            prev={prev}
            isOnline={on}
        />
    );
};