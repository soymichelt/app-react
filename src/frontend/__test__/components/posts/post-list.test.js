import React from 'react';
import {shallow} from 'enzyme';
import {create,} from 'react-test-renderer';
import {PostList} from './../../../posts/post-list';
import {ListPost} from './../../../__mocks__/list-post.mock';

describe("Testing PostList Component", () => {

    const wrapper = shallow(<PostList />);

    it('Renderizado del componente', () => {
        expect(wrapper.length).toEqual(1)
    })

    it("Render del tÃ­tulo del componente", () => {
        expect(wrapper.find('.header-title').text()).toEqual('Listado de Datos')
    });

    it("Render de la lista", () => {
        expect(wrapper.find('ul').length).toEqual(1)
    })
});

describe('Instantanea PostList Component', () => {
    it('Renderizado del componente', () => {
        expect(create(<PostList data={ListPost} />).toJSON()).toMatchSnapshot()
    })
})