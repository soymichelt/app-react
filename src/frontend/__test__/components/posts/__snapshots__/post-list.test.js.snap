// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instantanea PostList Component Renderizado del componente 1`] = `
Array [
  <h1
    className="header-title"
  >
    Listado de Datos
  </h1>,
  <ul
    id="list"
  >
    <li>
      Michel Tra√±a
    </li>
    <li>
      Roberto Tablada
    </li>
  </ul>,
  <button
    disabled={false}
  >
    Anterior
  </button>,
  <button
    disabled={false}
  >
    Siguiente
  </button>,
  <p>
    Offline
  </p>,
]
`;
